File: backend/src/controllers/bugs.py
from fastapi import APIRouter

from src.schemas.bug import BugIn, BugOut, BugUpdate
from src.services.bug import BugServiceDep

router = APIRouter(prefix="/bugs", tags=["bugs"])


@router.post("/", response_model=BugOut)
async def create_bug(
    bug_in: BugIn,
    bug_service: BugServiceDep,
):
    return await bug_service.create(bug_in)


@router.get("/", response_model=list[BugOut])
async def read_bugs(
    bug_service: BugServiceDep,
):
    return await bug_service.read_all()


@router.patch("/{bug_id}", response_model=BugOut)
async def update_bug(
    bug_id: int,
    bug_in: BugUpdate,
    bug_service: BugServiceDep,
):
    return await bug_service.update(bug_id, bug_in)


@router.delete("/{bug_id}")
async def delete_bug(
    bug_id: int,
    bug_service: BugServiceDep,
):
    await bug_service.delete(bug_id)
    return {"message": "Bug deleted successfully"}

File: FRONTEND/src/lib/api/bugs.ts
import { request } from "./client";
import type { Bug, BugIn, BugUpdate } from "./types";

export const bugsApi = {
  list: () => request<Bug[]>("/bugs/"),

  create: (data: BugIn) =>
    request<Bug>("/bugs/", { method: "POST", body: JSON.stringify(data) }),

  update: (id: number, data: BugUpdate) =>
    request<Bug>(`/bugs/${id}`, { method: "PATCH", body: JSON.stringify(data) }),

  delete: (id: number) =>
    request<void>(`/bugs/${id}`, { method: "DELETE" }),
};
